src/react-query/mutations/useAddToHistory.js

```diff
@@ -0,0 +1,14 @@
+import { useMutation } from "react-query";
+import { privateReq } from "../../axios";
+import { queryClient } from "../client";
+import { userKeys } from "../keyFactory";
+
+const addToHistory = async video => {
+  const res = await privateReq.post("history", { video });
```

Comment:
you can check here wether or not the user is authenticated or also you can just send the request regardless the user is authenticated or not because if the user is not authenticated the data is not gonna save